test/test.js

```diff
@@ -192,9 +586,209 @@ suite('GaiaHeader', function() {
+      Object.defineProperty(Element.prototype, 'clientWidth', this.clientWidth);
+    });
+
+    test('It doesn\'t call `.clientWidth` if title-start is used', function() {
+      this.container.innerHTML = `<gaia-header title-start="50">
+        <button></button>
+        <h1>Title</h1>
+      </gaia-header>`;
+
+      var element = this.container.firstElementChild;
+      var titleStart = element.titleStart;
+
+      sinon.assert.notCalled(this.clientWidth_spy.get, 'didn\'t query DOM');
+      assert.equal(titleStart, 50);
+    });
+
+    test('It doesn\'t call `.clientWidth` if title-end is used', function() {
+      this.container.innerHTML = `<gaia-header title-end="50">
+        <h1>Title</h1>
+        <button></button>
+      </gaia-header>`;
+
+      var element = this.container.firstElementChild;
+      var titleEnd = element.titleEnd;
+
+      sinon.assert.notCalled(this.clientWidth_spy.get, 'didn\'t query DOM');
+      assert.equal(titleEnd, 50);
+    });
+
+    test('It doesn\'t  call `.clientWidth` if their are no user buttons', function() {
+      this.container.innerHTML = `<gaia-header action="back">
+        <h1>Title</h1>
+      </gaia-header>`;
+
+      this.container.innerHTML = `<gaia-header>
+        <h1>Title</h1>
+      </gaia-header>`;
+
+      sinon.assert.notCalled(this.clientWidth_spy.get, 'didn\'t query DOM');
+    });
+
+    test('It uses `window.innerWidth` by default', function() {
+      this.container.innerHTML = `<gaia-header action="back">
+        <h1>Title</h1>
+      </gaia-header>`;
+
+      sinon.assert.called(this.innerWidth_spy);
+      sinon.assert.notCalled(this.clientWidth_spy.get, 'queried DOM');
+    });
+
+    test('It uses `gaiaHeader.clientWidth` when the `not-flush` attribute is used', function() {
+      this.container.innerHTML = `<gaia-header action="back" not-flush>
+        <h1>Title</h1>
+      </gaia-header>`;
+
+      sinon.assert.called(this.clientWidth_spy.get, 'queried DOM');
+    });
+  });
+
+  test('It works with multiple headers', function(done) {
+    this.container.innerHTML = `<gaia-header action="back">
+      <h1 class="title-1">Title 1</h1>
+      <h1 class="title-2" hidden>Title 2</h1>
+      <button></button>
+      <button></button>
+    </gaia-header>`;
+
+    var el = this.container.firstElementChild;
+    var title1 = el.querySelector('.title-1');
+    var title2 = el.querySelector('.title-2');
+
+    assert.equal(title1.style.MozMarginStart, '50px');
+    assert.equal(title2.style.MozMarginStart, '50px');
+
+    this.fontFit.reset();
+
+    title1.hidden = true;
+    title2.hidden = false;
+
+    waitForMutationObserver(() => {
+      sinon.assert.notCalled(this.fontFit, 'font-fit not required');
+      assert.equal(title1.style.MozMarginStart, '50px');
+      assert.equal(title2.style.MozMarginStart, '50px');
+      done();
+    });
+  });
+
+  /**
+   * Utils
+   */
+
+  function waitForMutationObserver(done) {
+    setTimeout(done, 50);
+  }
```

Comment:
maybe in a similar approach to https://github.com/gaia-components/gaia-header/blob/8fb74f6fb8a73b83b6509766e5a32421e202e47b/test/font_fit_test.js#L240-L266 ? (but more generic of course)